dan_layer/engine/src/runtime/working_state.rs

```diff
@@ -941,6 +943,14 @@ impl WorkingState {
             .ok_or(RuntimeError::NoActiveCallFrame)
     }
 
+    pub fn id_provider_no_call_frame(&self) -> Result<IdProvider<'_>, RuntimeError> {
+        Ok(IdProvider::new(
+            EntityId::default(),
```

Comment:
Not sure about this - entity id should be derived from the transaction hash, otherwise the component address will always be in shard 0 when DerivedComponentAddress is None